app/(candidate)/dashboard/election-result/components/ElectionResultPage.js

```diff
@@ -0,0 +1,162 @@
+'use client'
+import { useEffect, useState } from 'react'
+import { useRouter } from 'next/navigation'
+import { updateCampaign } from 'app/(candidate)/onboarding/shared/ajaxActions'
+import { useSnackbar } from 'helpers/useSnackbar'
+import { EVENTS, trackEvent } from 'helpers/analyticsHelper'
+import { apiRoutes } from 'gpApi/routes'
+import { clientFetch } from 'gpApi/clientFetch'
+import { useElectedOffice } from '@shared/hooks/useElectedOffice'
+import { LuTrophy, LuFrown } from 'react-icons/lu'
+import { useCampaign } from '@shared/hooks/useCampaign'
+import ResultOptionButton from './ResultOptionButton'
+
+const options = [
+  { key: 'won', label: 'I won my race', icon: <LuTrophy size={24} /> },
+  {
+    key: 'lost',
+    label: 'I lost my race',
+    icon: <LuFrown size={24} />,
+  },
+  // {
+  //   key: 'runoff',
+  //   label: 'Neither, I am in a run-off election',
+  //   icon: <LuMeh size={24} />,
+  // },
+]
+
+export default function ElectionResultPage({}) {
+  const router = useRouter()
+  const [campaign, setCampaign] = useCampaign()
+  const [selectedOption, setSelectedOption] = useState(null)
+  const { goals, details } = campaign || {}
+  const electionDate = details?.electionDate || goals?.electionDate
+  const officeName =
+    details?.office?.toLowerCase() === 'other'
+      ? details?.otherOffice
+      : details?.office
+
+  const { refreshElectedOffice } = useElectedOffice()
+  const { errorSnackbar } = useSnackbar()
+  const [requestState, setRequestState] = useState({
+    submitting: false,
+    error: false,
+  })
+
+  const createElectedOffice = async () => {
+    if (!electionDate) {
+      throw new Error('Invalid election date')
+    }
+
+    const response = await clientFetch(apiRoutes.electedOffice.create, {
+      electedDate: new Date(electionDate).toISOString().split('T')[0],
+    })
+
+    if (!response.ok) {
+      throw new Error('Failed to create elected office')
+    }
+    refreshElectedOffice()
+    return response.data
+  }
+
+  async function handleSelection(selection) {
+    setSelectedOption(selection)
+    setRequestState({ submitting: true, error: false })
+    try {
+      const wonGeneral = selection === 'won'
+
+      await updateCampaign([{ key: 'details.wonGeneral', value: wonGeneral }])
+
+      setCampaign((campaign) => ({
+        ...campaign,
+        details: {
+          ...campaign.details,
+          wonGeneral: wonGeneral,
+        },
+      }))
+
+      trackEvent(EVENTS.Candidacy.DidYouWinModalCompleted, {
+        status: selection,
+      })
+      // Create ElectedOffice if the user won the election
+      if (wonGeneral) {
+        await createElectedOffice()
+        router.push('/polls/welcome')
+      } else {
+        router.push('/dashboard/election-result/loss')
```

Comment:
No Back Navigation Protection
After selection, users are routed to either /polls/welcome or /dashboard/election-result/loss. If they hit the browser back button, they return to the election result page, potentially allowing double-submission.
Suggestion: Use router.replace() instead of router.push() to prevent back navigation:

```
router.replace('/polls/welcome')
router.replace('/dashboard/election-result/loss')
```